runtime/compiler/runtime/J9CodeCacheManager.cpp

```diff
@@ -76,6 +77,24 @@ J9::CodeCacheManager::initialize(bool useConsolidatedCache, uint32_t numberOfCod
    return self()->OMR::CodeCacheManager::initialize(useConsolidatedCache, numberOfCodeCachesToCreateAtStartup);
    }
 
+bool
+J9::CodeCacheManager::isAvailableMemorySafeForAllocation(size_t &requestedCodeCacheSize)
+   {
+   bool incomplete;
+   uint64_t freePhysicalMemoryB = getCompilationInfo(_jitConfig)->computeAndCacheFreePhysicalMemory(incomplete);
+
+   if (TR::Options::isAnyVerboseOptionSet(TR_VerbosePerformance,TR_VerboseCodeCache))
+   {
+   if (freePhysicalMemoryB != OMRPORT_MEMINFO_NOT_AVAILABLE)
+      TR_VerboseLog::writeLineLocked(TR_Vlog_CODECACHE, "Available freePhysicalMemory=%llu MB, requested codecache size=%lu MB", freePhysicalMemoryB >> 20, requestedCodeCacheSize);
+   }
+
+   if (freePhysicalMemoryB > (TR::Options::getSafeReservePhysicalMemoryValue() + requestedCodeCacheSize))
```

Comment:
Maybe we should not take an action if the information about memory availability is `incomplete`